ndcube/utils/wcs.py

```diff
@@ -197,18 +201,32 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    extra_coords_list = [] # initiating new list
+    pix_coords = [0] * len(item_) # Set up a list of pixel coords as input to all_pix2world
+    # Checking item_ slices for dropped axes if any
+    for i, slice_element in enumerate(item_):
+        if missing_axis[i] is False:
+            if (slice_element.stop - slice_element.start) == 1:
+                pix_coords[i] = slice_element.start 
+                # Enter pixel coordinate for this axis. 
+                # Since we are only dealing with independent axes the other axes can remain at 0.
+                real_world_coords = wcs.all_pix2world(*pix_coords, 0) 
+                # Unravel the arguments in the pix_coords array using the * prefix.
+                axis_name = pix_coords.wcs.ctype # Am working on this bit
```

Comment:
OK.  Let me know when this bit is ready for another review.

Reply:
Sure, give me some time to sit on it and I will get back to you soon. Will need to play around with it first on my own laptop.

Reply:
Hi @DanRyanIrish, regarding the `real_world_coords`, is it correct I should be getting a list of values? Is it like the example given in [https://docs.sunpy.org/projects/ndcube/en/stable/ndcube.html#extra-coordinates](https://docs.sunpy.org/projects/ndcube/en/stable/ndcube.html#extra-coordinates)?

Reply:
Yes.  You have to enter a pixel value(s) for each dimension to ```all_pix2world```.  This is why ```pix_coords``` is a list of values/arrays with the same length as number of dimensions.  Therefore you get back a list of real world coordinates for each dimension.  Just as you used the iterator ```i``` to implant the pixel value in the location of ```pix_coords``` corresponding to the dimension you were looking for, likewise the ith element of ```real_world_coords``` will give you the real world value of that same dimension.  Does that make sense?

So regarding the example in the ndcube docs, yes it is equivalent.  Except that ```NDCube.pixel_to_world``` returns ```Quantities```, i.e. the returned values have units attached.  ```all_pix2world``` only gives you the values and you have to know/figure out what the units are.  So you will have to manually construct a ```Quantity``` with the correct unit to enter into the ```dropped_coords``` variable.

Reply:
Yup, it makes sense now.

Reply:
No.  ```pix_coords``` is not an N-dimensional WCS object.  It is an iterable of array/pixel indices.  The old WCS object is the input variable, ```wcs```, while the sliced WCS object is ```new_wcs```.  Because the old WCS object ```wcs``` has not been sliced, it still contains the all the original axes.  Therefore we need to extract information on the dropped axes from it.    You have already done this for the real world coordinates where you have
```python
real_world_coords = wcs.all_pix2world(*pix_coords, 0)
```
(There is still a problem with this.  See point 2 in the more advanced comments below.)  Therefore similar to above, we get the CTYPE of the dropped axis by doing something like:
```python
ctype = wcs.wcs.ctype[i]
```
I admit that fact the ```wcs``` is also an attribute of the WCS object, ```wcs```, is a bit confusing.  (This actually only gets us halfway to what we want from an ```axis_name```.  However, it is sufficient for the specific issues raised by your question.  For where to go next once you've understood the above and made the necessary changes, see point 1 in the more advanced comments below.)

More advanced comments not required to answer your specific questions here.

1. Getting the ```axis_name```:
The reason I've named the variable above ```ctype```, rather than ```axis_name```, is that it doesn't give us a very intuitively understandable axis name.  This only gives us the CTYPE, which is a FITS-WCS abbreviation of the axis name.  To get a more understandable axis name from this, we use the International Virtual Observatory Alliance (IVOA) naming convention.  There is a dictionary, called ```wcs_ivoa_mapping``` in the ```ndcube.utils.wcs``` module (https://github.com/sunpy/ndcube/blob/master/ndcube/utils/wcs.py#L33) which tells you what IVOA axis name corresponds to what FITS-WCS CTYPE.  See how this dictionary is used in the ```NDCube.world_axis_physical_types``` code: https://github.com/sunpy/ndcube/blob/master/ndcube/ndcube.py#L233
So the axis name you want to enter into the ```dropped_coords``` variable is the IVOA axis name corresponding to the CTYPE.

2. Extracting Real World Coords for Dropped Axes
The problem with the way you've currently written ```real_world_coords = wcs.all_pix2world(*pix_coords, 0)```, is that ```wcs.all_pix2world``` performs the coordinate transformations of all axes at once.  This is why ```pix_coords``` is a ```list``` with one value/array of values for each dimension.  It should not be surprising therefore, that the output of ```wcs.all_pix2world``` includes the real world coordinates for each axis.  Therefore to extract the value/array of values for the dropped dimension, you must index ```real_world_coordinates``` with the iterator of the dropped dimension.  In our case, that is ```i```.  So another way of coding this is:
```python
dropped_axis_real_world_coords = wcs.all_pix2world(*pix_coords, 0)[i] # The dropped_axis_real_world_coords variable name is way too long but I've been super explicit here to help with clarity.
```

Reply:
Hi @DanRyanIrish, I am taking my time to go through your long reply a couple of times to make sure everything sinks in... Will get back to you soon with some more code commits.

Reply:
@DanRyanIrish, I see now I was stuck at the confusion of my identification of the `wcs` object with that of the `pix_coords`... and that I was inconsistent in my use of indexing within the `if` statement inside the for `loop` for collecting the relevant elements from the generated `ctype` and `pix_coords` lists of values/arrays. I have committed some more changes to this PR. Hopefully at least the `dropped_coords` definition code block now is at least self-consistent, if not functional.